Swap Caffe2 operator constructor to pass arguments by value

bwasti opened the pull request:
Summary:
allows instantiation of operator with arguments passed by move rather than explicit copies

per Sebastian's suggestion

Differential Revision: D13882416

smessmer reviewed (approved):
Looks good, thanks. As discussed, if `RunOperator()` was meant to stay, we should also change it towards taking its arguments by value. Since it's going to be replaced soon, but feel free to leave that one be.

bwasti committed:
Swap Caffe2 operator constructor to pass arguments by value (#16576)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/16576

allows instantiation of operator with arguments passed by move rather than explicit copies

per Sebastian's suggestion

Reviewed By: smessmer

Differential Revision: D13882416

fbshipit-source-id: 9756612c9db3f99e6d9b2206cd490a05b6fb6358
